src/main/java/com/exasol/releasedroid/repository/BaseRepository.java

```diff
@@ -156,16 +39,31 @@ public ReleaseLetter getReleaseLetter(final String version) {
 
     @Override
     public String getName() {
-        return this.fullName;
+        return this.repositoryGate.getName();
     }
 
     @Override
-    public String getVersion() {
-        if (getMavenPom().hasVersion()) {
-            return getMavenPom().getVersion();
-        } else {
-            throw new RepositoryException(ExaError.messageBuilder("E-RR-REP-5")
-                    .message("Cannot find the current version in the repository.").toString());
-        }
+    public Optional<String> getLatestTag() {
+        return this.repositoryGate.getLatestTag();
+    }
+
+    @Override
+    public String getSingleFileContentAsString(final String filePath) {
+        return this.repositoryGate.getSingleFileContentAsString(filePath);
+    }
+
+    @Override
+    public void updateFileContent(final String filePath, final String newContent, final String commitMessage) {
+        this.repositoryGate.updateFileContent(filePath, newContent, commitMessage);
+    }
+
+    @Override
+    public boolean isOnDefaultBranch() {
+        return this.repositoryGate.isOnDefaultBranch();
+    }
+
+    @Override
+    public String getBranchName() {
+        return this.repositoryGate.getBranchName();
```

Comment:
We need to think about all this delegation. Please create a ticket (unless there's one already) to address this situation